storage-node/src/cache/data_store_cache/memdisk/mod.rs

```diff
@@ -166,51 +242,66 @@ impl<R: DataStoreReplacer<MemDiskStoreReplacerKey, MemDiskStoreReplacerValue>> D
             None;
         // 1. If the mem_store is enabled, first try to write the data to memory.
         // Note: Only file which size < mem_max_file_size can be written to memory.
-        if let Some(mem_store) = &mut self.mem_store {
+        if let Some(mem_store) = &self.mem_store {
             loop {
                 match data_stream.next().await {
                     Some(Ok(bytes)) => {
                         bytes_mem_written += bytes.len();
                         // TODO: Need to write the data to network.
-                        if let Some((bytes_vec, _)) =
-                            mem_store.write_data(remote_location.clone(), bytes)
+                        if let Some((bytes_vec, _)) = mem_store
+                            .write()
+                            .await
+                            .write_data(remote_location.clone(), bytes)
```

Comment:
I am not sure which one is best, acquiring lock every time may have more concurrency since when polling next from S3, we can release lock for sometime and allow others to continue?
But the overhead is to grab lock every time. Maybe a TODO here and compare them via benchmark?

Reply:
I believe tokio's mutex/rwlock can automatically yield if we cannot make progress in the current task 🤔️

Reply:
A TODO is added for bench!